### ISSUE TITLE
Builtin keyword motivations and meanings

### ISSUE THREAD

**Issue by salotz** [tarsius/hl-todo#50]
Not an issue with the code base per se, but I was wondering what some of the keywords mean or are used for. Specifically: HOLD, NEXT, THEM, PROG, DONT, FAIL, and XXX+.

Also KLUDGE and HACK seem redundant, to me at least.

Also I have some ideas for some other ones if anyone's interested like STUB, OPT (for an optimization opportunity that was not used yet), GOTCHA and some others. Is there an article or mini-standard that anyone has or knows of? 

Would like to see a discussion on this, although likely the project would like to stay agnostic to any scheme.

**Comment by salotz** [comment]
PEP 350 has some guidelines: https://www.python.org/dev/peps/pep-0350/

**Comment by tarsius** [comment]
Many years ago I looked for a package like this one.  I found a few but none of them passed the initial smell test.  So I figured writing my own would be less work and that adding the following statement to the package commentary would be enough to manage expectations:

> See [this list](https://www.emacswiki.org/emacs/FixmeMode) on the Emacswiki for other packages that implement the same basic features, but which might also provide additional features that you might like, but which I don't deem necessary.

Somehow everyone ignored that.

And now more than a hundred commits later you want to discuss about the perfect set of keywords. There's nothing wrong about that, except that I don't.

The current set of keywords was picked because I had used that same set in Org-Mode before. It made more sense in Org-Mode than it makes in source code. Some users suggested additional keywords, some of which I added. I do not want to remove any existing keywords as there are probably people who use them.

I am open to add new individual keywords. I don't want to add complete sets of keywords (such as those needed by pep-0350) because ideally ~hl-todo-keyword-faces~ would be just one coherent set of keywords.  That is best achieved by:

``` emacs-lisp
(setq hl-todo-keyword-faces '(...))
```

While I am not interested in discussing why I picked these keywords and whether certain other keywords would be better and why, I would still encourage you to come up with a pep-0350 (or whatever) value for `hl-todo-keyword-faces`.  And to add that to the wiki.  Tell me if you want to do that; the wiki is currently disabled.

**Comment by salotz** [comment]
I did read that and I hesitated to open an issue. I really just wanted to make a little discussion and I didn't know where else to post this. Hence my qualification:

> Would like to see a discussion on this, although likely the project would like to stay agnostic to any scheme.

> Tell me if you want to do that; the wiki is currently disabled.

I would like to have somewhere to post these but I would take ownership of that somewhere else other than the wiki. Maybe just putting in a link or something to that and a few hints on how to customize your faces would be all I would propose.
